### PR TITLE
Add section about metrics reporting in README

### PR DESCRIPTION
**Pull request by aggarw13**
Add section of "Sending metrics to AWS IoT" in README to inform customers about an optional mechanism for them to report their platform and MQTT client metrics to AWS IoT

### CODE REVIEW

**Review by jainamt — DISMISSED** [review]

**Review by RichardBarry — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -21,6 +21,52 @@ and build the library with default configuration values, provide `MQTT_DO_NOT_US
 **OR**
 * Defining the `MQTT_DO_NOT_USE_CUSTOM_CONFIG` preprocessor macro for the library build.
 
+## Sending metrics to AWS IoT
+
+When establishing a connection with AWS IoT, users can optionally report the Operating System, Hardware Platform and MQTT client version information of their device to AWS. These metrics help AWS IoT improve security and provide better technical support. If users want to report this information, they can send a specially formatted string (see below) in the username field of the MQTT CONNECT packet.
```
Comment (tanmoysen) [inline README.md:26]: Second sentence --> "This information can help AWS IoT provide faster issue resolution and technical support."

Comment (aggarw13) [inline README.md:26]:
```suggestion
When establishing a connection with AWS IoT, users can optionally report the Operating System, Hardware Platform and MQTT client version information of their device to AWS. This information can help AWS IoT provide faster issue resolution and technical support. If users want to report this information, they can send a specially formatted string (see below) in the username field of the MQTT CONNECT packet.
```

Comment (aggarw13) [reply README.md]: Fixed

Comment (aggarw13) [reply README.md]: Fixed

**Review by tanmoysen — APPROVED** [review]

**Review by abhidixi11 — APPROVED** [review]

**Pull request merged**